cocotb/share/lib/vhpi/VhpiImpl.cpp

```diff
@@ -966,9 +966,13 @@ int VhpiImpl::deregister_callback(GpiCbHdl *gpi_hdl) {
 }
 
 void VhpiImpl::sim_end() {
-    sim_finish_cb->set_call_state(GPI_DELETE);
-    vhpi_control(vhpiFinish, vhpiDiagTimeLoc);
-    check_vhpi_error();
+    // Some sims do not seem to be able to deregister the end of sim callback
```

Comment:
Can you add an issue reference here to specify "some sims"? Otherwise we'll never know if we can remove the workaround.

Reply:
This came from the VPI. There is no listed rationale. Related commits: https://github.com/cocotb/cocotb/commit/8c435b0d3b7e855897a7c72e9b0aa8d005ac89e6 and https://github.com/cocotb/cocotb/commit/70dc1a6a39ad9d3991e5d20d9e7105bf5f59b4db.